src/software/ai/navigator/path_planner/theta_star_path_planner_test.cpp

```diff
@@ -105,24 +110,29 @@ TEST(TestThetaStarPathPlanner, test_theta_star_path_planner_blocked_dest)
     std::vector<Obstacle> obstacles = {Obstacle(
         Polygon({Point(3.5, 1), Point(2.5, 1), Point(2.5, -1), Point(3.5, -1)}))};
 
-    std::unique_ptr<PathPlanner> planner =
-        std::make_unique<ThetaStarPathPlanner>(field, obstacles);
-
+    std::unique_ptr<PathPlanner> planner = std::make_unique<ThetaStarPathPlanner>();
+    std::vector<Point> path_points;
 
-    auto path_points = planner->findPath(start, dest);
-
-    // We expect to find a path
-    ASSERT_TRUE(path_points);
+    try
+    {
+        path_points = std::get<std::vector<Point>>(
+            planner->findPath(start, dest, field, obstacles));
+    }
+    catch (const std::bad_variant_access&)
+    {
+        // not a valid path
+        EXPECT_TRUE(0);
```

Comment:
See above comment about using builtin gtest features for explicit failures. Same goes for all other cases of this.

Reply:
Done